### ISSUE TITLE
clean model file's gradients,only save the parameters 

### ISSUE THREAD

**Issue by zhu709654396** [ultralytics/ultralytics#1555]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

can i clean model file's gradients,only save the parameters  when i use model to infer 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@zhu709654396 I think the official weights and your trained model do not include any gradients.

**Comment by zhu709654396** [comment]
> 

                   from  n    params  module                                       arguments                                                                                                                                                                                      
  0                  -1  1       928  ultralytics.nn.modules.Conv                  [3, 32, 3, 2]                                                                                                                                                                                  
  1                  -1  1     18560  ultralytics.nn.modules.Conv                  [32, 64, 3, 2]                                                                                                                                                                                 
  2                  -1  1     29056  ultralytics.nn.modules.C2f                   [64, 64, 1, True]                                                                                                                                                                              
  3                  -1  1     73984  ultralytics.nn.modules.Conv                  [64, 128, 3, 2]                                                                                                                                                                                
  4                  -1  2    197632  ultralytics.nn.modules.C2f                   [128, 128, 2, True]                                                                                                                                                                            
  5                  -1  1    295424  ultralytics.nn.modules.Conv                  [128, 256, 3, 2]                                                                                                                                                                               
  6                  -1  2    788480  ultralytics.nn.modules.C2f                   [256, 256, 2, True]                                                                                                                                                                            
  7                  -1  1   1180672  ultralytics.nn.modules.Conv                  [256, 512, 3, 2]                                                                                                                                                                               
  8                  -1  1   1838080  ultralytics.nn.modules.C2f                   [512, 512, 1, True]                                                                                                                                                                              9                  -1  1    656896  ultralytics.nn.modules.SPPF                  [512, 512, 5]                                                                                                                                                                                  
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']                                                                                                                                                                           
 11             [-1, 6]  1         0  ultralytics.nn.modules.Concat                [1]                                                                                                                                                                                             12                  -1  1    591360  ultralytics.nn.modules.C2f                   [768, 256, 1]                                                                                                                                                                                  
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']                                                                                                                                                                           
 14             [-1, 4]  1         0  ultralytics.nn.modules.Concat                [1]                                                                                                                                                                                            
 15                  -1  1    148224  ultralytics.nn.modules.C2f                   [384, 128, 1]                                                                                                                                                                                  
 16                  -1  1    147712  ultralytics.nn.modules.Conv                  [128, 128, 3, 2]                                                                                                                                                                               
 17            [-1, 12]  1         0  ultralytics.nn.modules.Concat                [1]                                                                                                                                                                                            
 18                  -1  1    493056  ultralytics.nn.modules.C2f                   [384, 256, 1]                                                                                                                                                                                  
 19                  -1  1    590336  ultralytics.nn.modules.Conv                  [256, 256, 3, 2]                                                                                                                                                                               
 20             [-1, 9]  1         0  ultralytics.nn.modules.Concat                [1]                                                                                                                                                                                            
 21                  -1  1   1969152  ultralytics.nn.modules.C2f                   [768, 512, 1]                                                                                                                                                                                  
 22        [15, 18, 21]  1   2147008  ultralytics.nn.modules.Detect                [80, [128, 256, 512]]                                                                                                                                                                          
Model summary: 225 layers, 11166560 parameters, 11166544 gradients, 28.8 GFLOPs   

as above,11166560 parameters about 11M，11166544 gradients about 11M,so final model file about 22M,May I understand this correctly

**Comment by Laughing-q** [comment]
@zhu709654396 well parameters size is parameters size itself which is 11M and file size is file size itself which is 22M. These are two different size. So everything is fine here, nothing to worry about. :)

**Comment by zhu709654396** [comment]
> @zhu709654396参数大小是参数大小本身，它是 11M，文件大小是文件大小本身，它是 22M。这是两个不同的尺寸。所以这里一切都很好，没什么好担心的。:)

好吧，我以为做推理只用11M参数信息可以减少很多内存,还有一个问题,关于v8s.onnx 转rknn,报错不支持float16,但是使用32又过于大,可不可以量化成int精度呢

**Comment by Laughing-q** [comment]
@zhu709654396 我们目前没有内置支持rknn格式，这个问题可能与rknn那块有关，我不太了解

**Comment by zhu709654396** [comment]
> @zhu709654396我们的目标之前没有内部支持rknn格式，这个问题可能与rknn那个块有关，我不太明白

好的感谢回答

**Comment by wandaoyi** [comment]
> > @zhu709654396参数大小是参数大小本身，它是 11M，文件大小是文件大小本身，它是 22M。这是两个不同的尺寸。所以这里一切都很好，没什么好担心的。:)
> 
> 好吧，我以为做推理只用11M参数信息可以减少很多内存,还有一个问题,关于v8s.onnx 转rknn,报错不支持float16,但是使用32又过于大,可不可以量化成int精度呢

那，有没有，将 v8 模型 量化成 int8 的 onnx 方案呢？如果有的话，是否可以分享一下？谢谢。wandaoyi_dxq@qq.com
